airbyte-cdk/bulk/toolkits/load-db/src/test/kotlin/io/airbyte/cdk/load/toolkits/load/db/orchestration/TableNameMapFactoryTest.kt

```diff
@@ -0,0 +1,244 @@
+import org.junit.jupiter.api.Test
+
+class TableNameMapFactoryTest {
+
+    @Test
+    fun testThrowOnEmptyCatalog() {
+        val rawTableNameGenerator = mockk<RawTableNameGenerator>()
+        val finalTableNameGenerator = mockk<FinalTableNameGenerator>()
+        val columnNameGenerator = mockk<ColumnNameGenerator>()
+
+        val emptyCatalog = mockk<DestinationCatalog>()
+        every { emptyCatalog.streams } returns emptyList()
+
+        val factory =
+            TableCatalogFactory(
+                emptyCatalog,
+                rawTableNameGenerator,
+                finalTableNameGenerator,
+                columnNameGenerator
+            )
+
+        val exception = assertThrows(ConfigErrorException::class.java) { factory.get() }
+        assertTrue(exception.message!!.contains("catalog contained no streams"))
+    }
+
+    @Test
+    fun testTableNameCollision() {
+        val rawTableNameGenerator = mockk<RawTableNameGenerator>()
+        val finalTableNameGenerator = mockk<FinalTableNameGenerator>()
+        val columnNameGenerator = mockk<ColumnNameGenerator>()
+
+        val stream1 = createTestStream("foobarfoo", "a")
+        val stream2 = createTestStream("foofoo", "a")
+        val catalog = DestinationCatalog(listOf(stream1, stream2))
+
+        every { rawTableNameGenerator.getTableName(stream1.descriptor) } returns
+            TableName("raw_a", "raw_foofoo")
+        every { rawTableNameGenerator.getTableName(stream2.descriptor) } returns
+            TableName("raw_a", "raw_foofoo")
+
+        every { finalTableNameGenerator.getTableName(stream1.descriptor) } returns
+            TableName("final_a", "final_foofoo")
+        every { finalTableNameGenerator.getTableName(stream2.descriptor) } returns
+            TableName("final_a", "final_foofoo")
+        every { columnNameGenerator.getColumnName(any()) } answers
+            { call ->
+                val input = call.invocation.args[0] as String
+                ColumnNameGenerator.ColumnName(input, input)
+            }
+
+        val factory =
+            TableCatalogFactory(
+                catalog,
+                rawTableNameGenerator,
+                finalTableNameGenerator,
+                columnNameGenerator
+            )
+        val tableCatalog = factory.get()
+
+        val tableNames1 = tableCatalog[stream1]!!.tableNames
+        val tableNames2 = tableCatalog[stream2]!!.tableNames
+
+        val rawTableNames = setOf(tableNames1.rawTableName!!.name, tableNames2.rawTableName!!.name)
+        val finalTableNames =
+            setOf(tableNames1.finalTableName!!.name, tableNames2.finalTableName!!.name)
+        assertTrue(rawTableNames.contains("raw_foofoo"))
+        val rawTableNameWithHash = rawTableNames.first { it != "raw_foofoo" }
+        assertTrue(rawTableNameWithHash.startsWith("raw_foofoo_"))
+        assertTrue(
+            rawTableNameWithHash.substring("raw_foofoo_".length).matches(Regex("[0-9a-f]{3}"))
```

Comment:
nit: why do we need the regex? Our output should be deterministic, i.e. we can do an exact string match